File: sdkwork-ai-sdk-typescript/src/types/content-block.ts
import type { ImageFileContent } from './image-file-content';
import type { TextContent } from './text-content';

/** Message content block */
export interface ContentBlock {
  /** Content block type */
  type?: string;
  /** Text payload */
  text?: TextContent;
  /** Image file payload */
  image_file?: ImageFileContent;
}

File: sdkwork-ai-sdk-csharp/Models/ContentBlock.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Ai.Models
{
    public class ContentBlock
    {
        public string? Type { get; set; }
        public TextContent? Text { get; set; }
        public ImageFileContent? ImageFile { get; set; }
    }
}
